skyportal/handlers/api/candidate.py

```diff
@@ -898,13 +903,20 @@ def delete(self, candidate_id):
               application/json:
                 schema: Success
         """
-        c = Candidate.query.get(candidate_id)
-        if not (
-            self.associated_user_object.is_system_admin
-            or c.uploader_id == self.associated_user_object.id
-        ):
-            return self.error("Insufficient permissions.")
-        DBSession().delete(c)
+        cands = (
+            Candidate.query.filter(Candidate.obj_id == obj_id)
+            .filter(Candidate.filter_id == filter_id)
```

Comment:
Is there any enforcement/error checking that `filter_id` will indeed be passed as an integer. If not we probably need some logic here to ensure `obj_id` is a string and `filter_id` is an int.

Reply:
Updated to return an informative error message if an invalid `(obj_id, filter_id)` pair is provided.